container.go

```diff
@@ -514,21 +455,32 @@ func (container *container) registerCallback() error {
 func (container *container) unregisterCallback() error {
 	callbackNumber := container.callbackNumber
 
-	callbackMapLock.Lock()
-	defer callbackMapLock.Unlock()
+	callbackMapLock.RLock()
+	context := callbackMap[callbackNumber]
+	callbackMapLock.RUnlock()
 
-	handle := callbackMap[callbackNumber].handle
+	if context == nil {
+		return nil
+	}
+
+	closeChannels(context.channels)
```

Comment:
Out of curiosity, why do you closeChannels before the call to unregister instead of after it?